cnudie/retrieve.py

```diff
@@ -489,7 +489,8 @@ def version_lookup(
     default_ctx_repo: cm.OcmRepository=None,
     ocm_repository_lookup: OcmRepositoryLookup=None,
     oci_client: oc.Client=None,
-    default_absent_ok=True,
+    default_absent_ok: bool=True,
+    default_raise_on_error: bool=False,
```

Comment:
how about making this more generic by passing-in status-codes to be considered as error?